### PR TITLE
[Setup] Remove the dependency for typing_extensions

### LINKED ISSUE #1631: ModuleNotFoundError: No module named 'typing_extensions'
Good morning sky team,

An `ModuleNotFoundError` after update skypilot yesterday.

```{shell}
$ sky status -r
Traceback (most recent call last):
  File "/Users/hanqingliu/mambaforge/bin/sky", line 5, in <module>
    from sky.cli import cli
  File "/Users/hanqingliu/mambaforge/lib/python3.9/site-packages/sky/__init__.py", line 10, in <module>
    from sky import backends
  File "/Users/hanqingliu/mambaforge/lib/python3.9/site-packages/sky/backends/__init__.py", line 3, in <module>
    from sky.backends.cloud_vm_ray_backend import CloudVmRayBackend
  File "/Users/hanqingliu/mambaforge/lib/python3.9/site-packages/sky/backends/cloud_vm_ray_backend.py", line 27, in <module>
    from sky import cloud_stores
  File "/Users/hanqingliu/mambaforge/lib/python3.9/site-packages/sky/cloud_stores.py", line 14, in <module>
    from sky.data import data_utils
  File "/Users/hanqingliu/mambaforge/lib/python3.9/site-packages/sky/data/__init__.py", line 2, in <module>
    from sky.data.storage import Storage, StorageMode, StoreType
  File "/Users/hanqingliu/mambaforge/lib/python3.9/site-packages/sky/data/storage.py", line 8, in <module>
    from typing_extensions import TypeAlias
ModuleNotFoundError: No module named 'typing_extensions'
```

### PR DESCRIPTION
**Pull request by Michaelvll**
Fixes #1631

Tested (run the relevant ones):

- [x] `pip install .`

### CODE REVIEW

**Review by concretevitamin — APPROVED** [review]

File: sky/setup_files/setup.py [hunk 1]
```diff
@@ -83,6 +83,7 @@ def parse_readme(readme: str) -> str:
     'ray[default]>=1.9.0,<=2.2.0',
     'rich',
     'tabulate',
+    'typing-extensions',
```
Comment (romilbhardwaj) [inline sky/setup_files/setup.py:86]: Can we remove our dependency? Like L31 and L32 in storage.py, can we change L33 and L34 to:
Before
```
StorageHandle = Any
StorageStatus = global_user_state.StorageStatus
Path: TypeAlias = str
SourceType: TypeAlias = Union[Path, List[Path]]
```

After
```
StorageHandle = Any
StorageStatus = global_user_state.StorageStatus
Path = str
SourceType = Union[Path, List[Path]]
```

**Review by romilbhardwaj — CHANGES_REQUESTED** [review]

File: sky/setup_files/setup.py (hunk 1 above)
Comment (romilbhardwaj) [reply sky/setup_files/setup.py:86]: mypy (format.py) appears to pass with this change

Comment (Michaelvll) [reply sky/setup_files/setup.py]: Ahh, great catch! Removed the dependency. Thanks!

**Review by romilbhardwaj — APPROVED** [review]
Awesome! Once the PR title is updated we should be good to merge

**Pull request merged**